File: apps/runner-ch/pkg/models/metrics.go
// Copyright 2025 Daytona Platforms Inc.
// SPDX-License-Identifier: AGPL-3.0

package models

import "time"

// SystemMetrics holds system-level metrics for the runner
type SystemMetrics struct {
	CPUUsage        float64   `json:"cpuUsage"`        // CPU usage percentage
	RAMUsage        float64   `json:"ramUsage"`        // RAM usage percentage
	DiskUsage       float64   `json:"diskUsage"`       // Disk usage percentage
	TotalCPU        int64     `json:"totalCPU"`        // Total CPU cores
	TotalRAMGiB     float64   `json:"totalRAMGiB"`     // Total RAM in GiB
	TotalDiskGiB    float64   `json:"totalDiskGiB"`    // Total disk in GiB
	AllocatedCPU    int64     `json:"allocatedCPU"`    // Allocated CPU cores
	AllocatedMemory int64     `json:"allocatedMemory"` // Allocated memory in GiB
	AllocatedDisk   int64     `json:"allocatedDisk"`   // Allocated disk in GB
	SnapshotCount   int64     `json:"snapshotCount"`   // Number of snapshots
	LastUpdated     time.Time `json:"lastUpdated"`     // When metrics were last updated
}

File: apps/dashboard/src/constants/metrics.ts
/*
 * Copyright Daytona Platforms Inc.
 * SPDX-License-Identifier: AGPL-3.0
 */

export const METRIC_DISPLAY_NAMES: Record<string, string> = {
  'daytona.sandbox.cpu.utilization': 'CPU Utilization',
  'daytona.sandbox.cpu.limit': 'CPU Limit',
  'daytona.sandbox.memory.utilization': 'Memory Utilization',
  'daytona.sandbox.memory.usage': 'Memory Usage',
  'daytona.sandbox.memory.limit': 'Memory Limit',
  'daytona.sandbox.filesystem.utilization': 'Disk Utilization',
  'daytona.sandbox.filesystem.usage': 'Disk Usage',
  'daytona.sandbox.filesystem.total': 'Disk Total',
  'daytona.sandbox.filesystem.available': 'Disk Available',
  'system.memory.utilization': 'System Memory Utilization',
}

export function getMetricDisplayName(metricName: string): string {
  return METRIC_DISPLAY_NAMES[metricName] ?? metricName.replace(/^daytona\.sandbox\./, '')
}
